core/src/test/java/io/questdb/griffin/engine/table/AsyncFilteredRecordCursorFactoryTest.java

```diff
@@ -405,6 +404,47 @@ public void testSymbolRegexBindVariableFilter() throws Exception {
         });
     }
 
+    @Test
+    public void testFaultToleranceNPE() throws Exception {
+        withPool0((engine, compiler, sqlExecutionContext) -> {
+            compiler.compile("create table x as (select rnd_double() a, rnd_symbol('a', 'b', 'c') s, timestamp_sequence(20000000, 100000) t from long_sequence(20)) timestamp(t) partition by hour", sqlExecutionContext);
+            final String sql = "select * from x where npe()";
+            try {
+                try (final RecordCursorFactory factory = compiler.compile(sql, sqlExecutionContext).getRecordCursorFactory()) {
+                    try (final RecordCursor cursor = factory.getCursor(sqlExecutionContext)) {
+                        while (cursor.hasNext()) {} // drain cursor until exception
+                        Assert.fail();
+                    }
+                }
+            } catch (Throwable e) {
+                MatcherAssert.assertThat(e.getMessage(), StringContains.containsString("timeout, query aborted"));
+            }
+        }, 4, 0); // sharedWorkerCount < workerCount
+    }
+
+    @Test
+    public void testFaultToleranceWrongSharedWorkerConfiguration() throws Exception {
+        withPool0((engine, compiler, sqlExecutionContext) -> {
+            compiler.compile("create table x as (select rnd_double() a, rnd_symbol('a', 'b', 'c') s, timestamp_sequence(20000000, 100000) t from long_sequence(20000)) timestamp(t) partition by hour", sqlExecutionContext);
+            final String sql = "select sum(a) from x where s='a'";
+            try {
+                // !!! test depends on thread scheduling
+                // should return the expected result or fail with a CairoException
```

Comment:
Could you describe the situation when this test may lead to a `CairoException` on this branch? Probably I'm missing something.

Reply:
It's actually test for a misconfiguration of workerCount.
More specifically, to test the assertion or NPE from this call:
```
            public function getFilter(int workerId) {
                if (workerId == -1 || perWorkerFilters == null) {
                    return filter;
                }
                return perWorkerFilters.getQuick(workerId);
            }
```
By setting the sharedWorkerCount to 0, we are effectively creating empty perWorkerFilters array.
But there is a way to schedule workerId to always be -1.


This test should fails or pass but never to crash